typescript/helloworld/src/app/app.ts

```diff
@@ -1,27 +1,40 @@
 import { BigNumber, ethers } from 'ethers';
 
-import { TypedListener } from '@abacus-network/core/dist/common';
 import {
   AbacusApp,
+  AbacusCore,
+  ChainMap,
   ChainName,
   ChainNameToDomainId,
+  MultiProvider,
   Remotes,
 } from '@abacus-network/sdk';
 
-import { ReceivedHelloWorldEvent } from '../types/contracts/HelloWorld';
-
 import { HelloWorldContracts } from './contracts';
 
+type Counts = {
+  sent: number;
+  received: number;
+};
+
 export class HelloWorldApp<
   Chain extends ChainName = ChainName,
 > extends AbacusApp<HelloWorldContracts, Chain> {
+  constructor(
+    public readonly core: AbacusCore<Chain>,
+    contractsMap: ChainMap<Chain, HelloWorldContracts>,
+    multiProvider: MultiProvider<Chain>,
+  ) {
+    super(contractsMap, multiProvider);
+  }
+
   async sendHelloWorld<From extends Chain>(
     from: From,
     to: Remotes<Chain, From>,
     message: string,
     value: BigNumber,
-    receiveHandler?: TypedListener<ReceivedHelloWorldEvent>,
-  ): Promise<ethers.ContractReceipt> {
+    afterSend?: (receipt: ethers.ContractReceipt) => void,
```

Comment:
not super important but this is quite a breaking change to the semantic of this parameter

Reply:
Yes, the old one was not used and the new one is. LMK if I am making a mistake on the assumption the old one was unused.

Reply:
Sorry for this drive-by comment/review, but on first impression, this callback pattern feels out of place for me? Why not continue to return the receipt and then have the caller just call the callback themselves after awaiting the send?